### ISSUE TITLE
Can't save posts as html

### ISSUE THREAD

**Issue by Raemon** [LessWrong2/Lesswrong2#1029]
https://app.intercom.io/a/apps/wtb8z7sj/inbox/inbox/all/conversations/18760392814

Hi! For some reason saving posts as html files for further reading doesn’t work in the new version. When opened, a file shows the text for a brief second and then says “Sorry, we couldn't find what you were looking for.”

**Comment by Raemon** [comment]
Also the code is really obfuscated and therefore hard to parse (if I, say, want to make an FB2 book out of some posts to read them in FBReader, not in the browser).

**Comment by jimrandomh** [comment]
The code isn't obfuscated, but it is compiled; our toolchain (Meteor) first does compilation to support older browsers, then minifies it to speed up load time. If you want to get data for use in another program, one possibility would be to use our GraphQL API.

What happens when you save a post as HTML from the browser is that it saves the server-side-rendered version, but then the page's Javascript tries to update the render, load comments and notifications, etc, and finds itself running in the wrong environment and gets confused. In particular, I suspect it's looking at the URL it's at, it sees the wrong path and interprets it as a navigation event having happened, and tries to navigate. The solution would be to add some code early in the process of Javascript initialization which sanity-checks the environment, and stops running scripts if things don't look right.
